matrix/helper_functions.hpp

```diff
@@ -55,7 +55,7 @@ Type wrap(Type x, Type low, Type high) {
     }
 
     const Type range = high - low;
-    const Type inv_range = Type(1) / range; // should evaluate at compile time, multiplies below at runtime
+    const double inv_range = 1.0 / range; // should evaluate at compile time, multiplies below at runtime
```

Comment:
This could massively increase the cost of wrapping for `float`on Cortex-4 chips which don't support `double`. Could you check maybe with [godbolt](https://godbolt.org/) whether this still eliminates the temporaries for floats with ARM GCC 8? Otherwise we might need a template specialization for integer types.

Reply:
Probably true, I'll have a look on godbolt tomorrow.